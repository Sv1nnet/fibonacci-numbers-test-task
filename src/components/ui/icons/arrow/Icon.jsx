import React from 'react';
import PropTypes from 'prop-types';

import './icon.scss';

const Icon = ({ className }) => (
  <span className={`arrow-icon ${className}`} />
);

Icon.propTypes = {
  className: PropTypes.string.isRequired,
};

Icon.defaultProps = {
  className: '',
};

export default Icon;
