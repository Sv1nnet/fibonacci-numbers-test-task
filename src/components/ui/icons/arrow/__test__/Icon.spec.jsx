import React from 'react';
import {
  it,
  describe,
  expect,
} from '@jest/globals';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'materialize-css';
import toJson from 'enzyme-to-json';
import Icon from '../Icon';

configure({ adapter: new Adapter() });

describe('Testing Icon component', () => {
  it('Should match snapshot with empty props', () => {
    const component = shallow(<Icon />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Should match snapshot with extra class passed to props', () => {
    const testClass = 'test-class';
    const component = shallow(<Icon className={testClass} />);

    expect(component.find('span').prop('className').includes(testClass)).toBe(true);
    expect(toJson(component)).toMatchSnapshot();
  });
});
