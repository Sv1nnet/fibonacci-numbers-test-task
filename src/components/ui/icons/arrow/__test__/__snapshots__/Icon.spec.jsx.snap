// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Icon component Should match snapshot with empty props 1`] = `
<span
  className="arrow-icon "
/>
`;

exports[`Testing Icon component Should match snapshot with extra class passed to props 1`] = `
<span
  className="arrow-icon test-class"
/>
`;
